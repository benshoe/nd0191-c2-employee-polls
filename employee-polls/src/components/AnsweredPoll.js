import PropTypes from "prop-types";

const AnsweredPoll = ({question, users, answeredOne}) => {

    const usersAnsweredOne = question.optionOne.votes.length;
    const usersAnsweredTwo = question.optionTwo.votes.length;
    const numberOfUsers = Object.keys(users).length;
    const answeredTwo = !answeredOne;

    return (
        <div className="poll-radio">
            <table>
                <thead>
                <tr>
                    <td>You answered</td>
                    <td>Option</td>
                    <td>Number of voters</td>
                    <td>Percentage voted</td>
                </tr>
                </thead>
                <tbody>
                <tr>
                    <td>{answeredOne && <div>X</div>}</td>
                    <td>{question.optionOne.text}</td>
                    <td>{question.optionOne.votes.length}</td>
                    <td>{parseFloat(usersAnsweredOne / numberOfUsers) * 100}%</td>
                </tr>
                <tr>
                    <td>{answeredTwo && <div>X</div>}</td>
                    <td>{question.optionTwo.text}</td>
                    <td>{question.optionTwo.votes.length}</td>
                    <td>{(usersAnsweredTwo / numberOfUsers) * 100}%</td>
                </tr>
                </tbody>
            </table>
        </div>)

}

export default AnsweredPoll;

AnsweredPoll.propTypes = {
    question: PropTypes.object.isRequired,
    users: PropTypes.object.isRequired,
    answeredOne: PropTypes.bool.isRequired,
}
