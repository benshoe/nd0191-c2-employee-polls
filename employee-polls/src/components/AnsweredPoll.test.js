import renderer from 'react-test-renderer';
import AnsweredPoll from './AnsweredPoll';
import React from 'react';

describe('tests', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<AnsweredPoll answeredOne={true} question={
                {
                    optionOne: {
                        votes: ['sarahedo'],
                        text: 'Build our new application with Javascript',
                    },
                    optionTwo: {
                        votes: [],
                        text: 'Build our new application with Typescript'
                    }
                }
            } users={
                {
                    mtsamis: {
                        id: 'mtsamis'
                    }
                }} />)
            .toJSON();
        expect(tree).toMatchSnapshot({});
    });
});
