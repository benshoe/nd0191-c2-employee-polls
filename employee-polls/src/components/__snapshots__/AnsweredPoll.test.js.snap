// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests renders correctly 1`] = `
Object {
  "children": Array [
    <table>
      <thead>
        <tr>
          <td>
            You answered
          </td>
          <td>
            Option
          </td>
          <td>
            Number of voters
          </td>
          <td>
            Percentage voted
          </td>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <div>
              X
            </div>
          </td>
          <td>
            Build our new application with Javascript
          </td>
          <td>
            1
          </td>
          <td>
            100
            %
          </td>
        </tr>
        <tr>
          <td />
          <td>
            Build our new application with Typescript
          </td>
          <td>
            0
          </td>
          <td>
            0
            %
          </td>
        </tr>
      </tbody>
    </table>,
  ],
  "props": Object {
    "className": "poll-radio",
  },
  "type": "div",
}
`;
